testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_cbl_log_redaction.py

```diff
@@ -0,0 +1,166 @@
+import subprocess
+import pytest
+
+from keywords.MobileRestClient import MobileRestClient
+from CBLClient.Replication import Replication
+from CBLClient.Authenticator import Authenticator
+from libraries.provision.ansible_runner import AnsibleRunner
+
+from libraries.testkit import cluster
+
+
+@pytest.mark.sanity
+@pytest.mark.listener
+@pytest.mark.replication
+@pytest.mark.parametrize("password", [
+    ("auto-password"),
+    ("auto password"),
+    ("validpassword"),
+])
+def test_mask_password_in_logs(params_from_base_test_setup, password):
+    """
+        @summary:
+        1. Create CBL DB and create bulk doc in CBL
+        2. Configure replication to Sg with basic authentication
+        3. Replicate docs to SG
+        4. Verify password is masked in cbl logs
+    """
+    sg_db = "db"
+
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+    log_file = params_from_base_test_setup["log_filename"]
+    liteserv_platform = params_from_base_test_setup["liteserv_platform"]
+
+    num_cbl_docs = 5
+    # Cannot run on iOS as there is no support in xcode to grab cbl logs
+    if sync_gateway_version < "2.0.0" or liteserv_platform.lower() == "ios":
+        pytest.skip('This test cannnot run with sg version below 2.0 and if platform is iOS')
+
+    channels = ["ABC"]
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+    # Clean up tmp logs before test runs
+    if liteserv_platform == "net-msft":
+        delete_tmp_logs()
+    replicator = Replication(base_url)
+    authenticator = Authenticator(base_url)
+
+    # One shot replication
```

Comment:
this comment should be at appropriate place

Reply:
removed the comment. Yeah we can add before replication method, but we are focusing more on verifying masking password , so I removed this.